src/main/kotlin/com/exactpro/th2/conn/amqp/BoxMain.kt

```diff
@@ -68,59 +68,62 @@ fun main(args: Array<String>) {
         )
 
         val rawRouter: MessageRouter<RawMessageBatch> = factory.messageRouterRawBatch
-
-        val publisher = MessagePublisher(
-            configuration.sessionAlias,
-            configuration.drainIntervalMills,
-            rawRouter
-        )
-        resources += publisher
-
-        val service: ConnService = ConnServiceImpl(
-            parameters = configuration.parameters,
-            onMessage = publisher::onMessage,
-            onEvent = { event ->
-                eventRouter.safeSend(event, rootEvent.id)
-            }
-        )
-        resources += service
-
-        rawRouter.subscribeAll { _, rawBatch ->
-            rawBatch.messagesList.forEach { msg ->
-                msg.runCatching(service::send).onFailure {
-                    eventRouter.safeSend(
-                        Event.start().endTimestamp()
-                            .status(Event.Status.FAILED)
-                            .type("SendError")
-                            .name("Cannot send message: ${msg.metadata}")
-                            .apply {
-                                var ex: Throwable? = it
-                                while(ex != null) {
-                                    bodyData(EventUtils.createMessageBean(ex.message))
-                                    ex = ex.cause
-                                }
-                            },
-                        if (msg.hasParentEventId()) msg.parentEventId.id else rootEvent.id
-                    )
-                }.onSuccess {
-                    if (configuration.enableMessageSendingEvent) {
-                        eventRouter.safeSend(
-                            Event.start().endTimestamp()
-                                .status(Event.Status.PASSED)
-                                .type("Message")
-                                .name("Message was sent:  ${msg.metadata.id.sequence}")
-                                .apply {
-                                    messageID(msg.metadata.id)
-                                },
-                            if (msg.hasParentEventId()) msg.parentEventId.id else rootEvent.id
-                        )
+        configuration.parameters.forEach{connParameters ->
+            run {
+                val publisher = MessagePublisher(
+                    connParameters.sessionAlias,
+                    configuration.drainIntervalMills,
+                    rawRouter
+                )
+                resources += publisher
+
+                val service: ConnService = ConnServiceImpl(
+                    parameters = connParameters,
+                    onMessage = publisher::onMessage,
+                    onEvent = { event ->
+
+                        eventRouter.safeSend(event, rootEvent.id)
+                    }
+                )
+                resources += service
+
+                rawRouter.subscribeAll { _, rawBatch ->
+                    rawBatch.messagesList.forEach { msg ->
```

Comment:
Conn will send all incoming (from MQ) messages to each sessions. Each session should work only messages compatible by session alias specified in msg.metadata.id.connectionId.sessionAlias.

Also, conn can call the subscribeAll method only once and organize message routing between services in the lambda